--- spi_lld.c	2012-06-18 19:14:28.000000000 +0200
+++ /home/formica/chibios/chibios/os/hal/platforms/STM32/spi_lld.c	2013-04-10 12:21:44.096847872 +0200
@@ -294,10 +294,16 @@
   }
   /* SPI setup and enable.*/
   spip->spi->CR1  = 0;
+  #if !defined(STM32_SPI_SLAVE_MODE)
   spip->spi->CR1  = spip->config->cr1 | SPI_CR1_MSTR | SPI_CR1_SSM |
                     SPI_CR1_SSI;
   spip->spi->CR2  = SPI_CR2_SSOE | SPI_CR2_RXDMAEN | SPI_CR2_TXDMAEN;
   spip->spi->CR1 |= SPI_CR1_SPE;
+  #else
+  spip->spi->CR1 = spip->config->cr1 | SPI_CR1_RXONLY; //| SPI_CR1_DFF;
+  spip->spi->CR2 = SPI_CR2_RXDMAEN; //| SPI_CR2_RXNEIE;
+  spip->spi->CR1 |= SPI_CR1_SPE;
+  #endif
 }
 
 /**
